packages/synapse-react-client/src/components/widgets/query-filter/EnumFacetFilter/EnumFacetFilterUI.tsx

```diff
@@ -0,0 +1,277 @@
+import { Collapse, IconButton, TextField, Tooltip } from '@mui/material'
+import React, { useMemo, useState } from 'react'
+import { FacetColumnResultValueCount } from '@sage-bionetworks/synapse-types'
+import IconSvg from '../../../IconSvg/IconSvg'
+import { Checkbox } from '../../Checkbox'
+import { FacetFilterHeader } from '../FacetFilterHeader'
+import { EnumFacetFilterOption } from './EnumFacetFilterOption'
+import EnumFacetFilterDropdown from './EnumFacetFilterDropdown'
+
+const MAX_ENUMERATION_VALUES_TO_SHOW = 5
+
+export type RenderedFacetValue = FacetColumnResultValueCount & {
+  /* Text displayed for the option, and also used to filter in search */
+  displayText: string
+}
+
+export type EnumFacetFilterUIProps = {
+  /* The title of the faceted column to be displayed */
+  facetTitle: string
+  /* List of all facet values and information associated with each value */
+  facetValues: RenderedFacetValue[]
+  /* Whether the 'All' checkbox should be selected. */
+  allIsSelected: boolean // allIsSelected cannot be derived from facetValues[*].isSelected. See PORTALS-2680
+  /* The EnumFacetFilter can be rendered as a collapsible panel or a dropdown. Default is `'Collapsible'` */
+  containerAs?: 'Collapsible' | 'Dropdown'
+  /* If `containerAs` is a dropdown, the dropdown can be either an IconButton or a SelectBox. Default is `'Icon'` */
+  dropdownType?: 'Icon' | 'SelectBox'
+  /* If true, hides the button to collapse the panel (only shown if `containerAs` is 'Collapsible'). Default is false.*/
+  hideCollapsible?: boolean
+  /* Called when an unselected value is selected */
+  onAddValueToSelection: (value: string) => void
+  /* Called when a selected value is unselected */
+  onRemoveValueFromSelection: (value: string) => void
+  /* Called when all values should be unselected (e.g. 'All' is clicked) */
+  onRemoveAllFacetSelections: () => void
+}
+
+function removeWhitespace(value: string): string {
+  return value.replace(/\s/g, '').toLowerCase()
+}
+
+export default function EnumFacetFilterUI(props: EnumFacetFilterUIProps) {
```

Comment:
Given a list of `RenderedFacetValue` (and all of the other props above), render the UI for the EnumFacetFilter. Most importantly, this is decoupled from `QueryWrapper`.

Some actions that this handles include
- Collapse/expand
- Truncating/expanding the list of facets to show
- Client-side search of facet values.